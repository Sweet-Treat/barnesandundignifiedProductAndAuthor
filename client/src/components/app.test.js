import React from 'react';
import { shallow } from 'enzyme';
import App from './app';

const exampleRecord =
  {
    'isbn13': '9781571311931',
    'author': 'Aimee Nezhukumatathil',
    'bookTitle': 'World of Wonders: In Praise of Fireflies, Whale Sharks, and Other Astonishments (B&N Exclusive Gift Edition)',
    'publisherName': 'Milkweed Editions',
    'publisherLink': '/publisher',
    'publicationDate': '12/02/2020',
    'series': '',
    'bookCategory': 'Nonfiction',
    'ageRange': '',
    'pages': 184,
    'salesRank': 311,
    'format': '',
    'editionDescription': '',
    'productDimensions': '6.50(w) x 1.50(h) x 9.50(d)'
  };

jest.mock('axios', () => {

  return {
    get: jest.fn(() => Promise.resolve({data: exampleRecord} )),
  };
});

const axios = require('axios');

it('fetch book record on #componentDidMount', async (done) => {
  const app = await shallow(<App isbn13={'9781571311931'}/>);
  expect(axios.get).toHaveBeenCalled();
  expect(axios.get).toHaveBeenCalledWith('http://localhost:5001/products/9781571311931');
  // console.log(app.state());

  expect(app.state()).toHaveProperty('book', exampleRecord);
  done();

  // Promises way:
  // app
  //   .instance()
  //   .componentDidMount()
  //   .then(() => {
  //     expect(axios.get).toHaveBeenCalled();
  //     expect(axios.get).toHaveBeenCalledWith('http://localhost:5001/products/9781571311931');
  //     expect(app.state()).toHaveProperty('book', exampleRecord);
  //     done();
  //   });
});