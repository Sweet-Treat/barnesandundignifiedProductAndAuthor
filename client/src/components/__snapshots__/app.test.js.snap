// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
"<App isbn13=\\"9781571311931\\">
  <div className=\\"tabsBox\\">
    <Tabs>
      <div className=\\"tabs\\">
        <ol className=\\"tab-list\\">
          <Tab activeTab=\\"Product Details\\" label=\\"Product Details\\" onClick={[Function: bound onClickTabItem]}>
            <li className=\\"tab-list-item tab-list-active\\" onClick={[Function: bound onClick]}>
              Product Details
            </li>
          </Tab>
          <Tab activeTab=\\"Product Details\\" label=\\"Author\\" onClick={[Function: bound onClickTabItem]}>
            <li className=\\"tab-list-item\\" onClick={[Function: bound onClick]}>
              Author
            </li>
          </Tab>
        </ol>
        <div className=\\"tab-content\\">
          <ProductDetails record={{...}}>
            <div className=\\"wrapper\\">
              <div className=\\"slide\\">
                <h2>
                  Product Details
                </h2>
                <table>
                  <tbody>
                    <tr>
                      <th>
                        ISBN-13: 
                      </th>
                      <td />
                    </tr>
                    <tr>
                      <th>
                        Publisher: 
                      </th>
                      <td>
                        <a href=\\"\\" />
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Publication Date: 
                      </th>
                      <td>
                        Invalid Date
                      </td>
                    </tr>
                    <tr>
                      <th>
                        Pages: 
                      </th>
                      <td />
                    </tr>
                    <tr>
                      <th>
                        Sales Rank: 
                      </th>
                      <td />
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </ProductDetails>
        </div>
      </div>
    </Tabs>
  </div>
</App>"
`;
